Fixes test/parallel/test-node-run.js subtest.

Note that this techinically depends on an impurity during build, see
also https://github.com/NixOS/nix/issues/6081. Currently stdenv allows
/bin/sh as an impure host dependency on Darwin, so we are guaranteed to
have it to certain degree, although it is very hacky.

--- a/test/fixtures/run-script/node_modules/.bin/positional-args
+++ b/test/fixtures/run-script/node_modules/.bin/positional-args
@@ -1,3 +1,3 @@
-#!/bin/bash
+#!/bin/sh
 echo "Arguments: '$@'"
 echo "The total number of arguments are: $#"
